components/scream/CMakeLists.txt

```diff
@@ -4,8 +4,6 @@ if (NOT DEFINED PROJECT_NAME)
   set(SCREAM_CIME_BUILD FALSE)
 else()
   set(SCREAM_CIME_BUILD TRUE)
-  set (SCREAM_SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR})
```

Comment:
This might come down to taste, but why removing these? As it is now, scream does not call `project()` in a CIME build, so these two won't get declared in a CIME build (`project()` sets them for you). They are handy variables, since they are [standard cmake vars](https://cmake.org/cmake/help/latest/variable/PROJECT-NAME_SOURCE_DIR.html) (if `project()` is called). Users are 'used' to see `<PROJECT-NAME>_SOURCE_DIR` and `<PROJECT-NAME>_BINARY_DIR`. On the other hand, `SCREAM_BIN_DIR` or `SCREAM_BASE_DIR` are less obvious.

Btw, imho, we should _always_ call project (cmake encourages it), so that they are always defined.

Reply:
If you look around lines 94-98, variables with identical purpose were already being set. This just removes the duplication.

Reply:
Oh, I see what you mean. Maybe we should rename the 94-98 ones to use the standard convention? At least until we refactor things to call `project`?

Reply:
Yeah, that was my idea. But if you think we might do another PR soon, where we call project no matter what, then it can wait.